beacon-chain/p2p/interfaces.go

```diff
@@ -21,95 +21,103 @@ import (
-	BroadcastLightClientOptimisticUpdate(ctx context.Context, update interfaces.LightClientOptimisticUpdate) error
-	BroadcastLightClientFinalityUpdate(ctx context.Context, update interfaces.LightClientFinalityUpdate) error
-	BroadcastDataColumn(root [fieldparams.RootLength]byte, columnSubnet uint64, dataColumnSidecar *ethpb.DataColumnSidecar, peersChecked ...chan<- bool) error
-}
-
-// SetStreamHandler configures p2p to handle streams of a certain topic ID.
-type SetStreamHandler interface {
-	SetStreamHandler(topic string, handler network.StreamHandler)
-}
-
-// PubSubTopicUser provides way to join, use and leave PubSub topics.
-type PubSubTopicUser interface {
-	JoinTopic(topic string, opts ...pubsub.TopicOpt) (*pubsub.Topic, error)
-	LeaveTopic(topic string) error
-	PublishToTopic(ctx context.Context, topic string, data []byte, opts ...pubsub.PubOpt) error
-	SubscribeToTopic(topic string, opts ...pubsub.SubOpt) (*pubsub.Subscription, error)
-}
-
-// ConnectionHandler configures p2p to handle connections with a peer.
-type ConnectionHandler interface {
-	AddConnectionHandler(f func(ctx context.Context, id peer.ID) error,
-		j func(ctx context.Context, id peer.ID) error)
-	AddDisconnectionHandler(f func(ctx context.Context, id peer.ID) error)
-	connmgr.ConnectionGater
-}
-
-// SenderEncoder allows sending functionality from libp2p as well as encoding for requests and responses.
-type SenderEncoder interface {
-	EncodingProvider
-	Sender
-}
-
-// EncodingProvider provides p2p network encoding.
-type EncodingProvider interface {
-	Encoding() encoder.NetworkEncoding
-}
-
-// PubSubProvider provides the p2p pubsub protocol.
-type PubSubProvider interface {
-	PubSub() *pubsub.PubSub
-}
-
-// PeerManager abstracts some peer management methods from libp2p.
-type PeerManager interface {
-	Disconnect(peer.ID) error
-	PeerID() peer.ID
-	Host() host.Host
-	ENR() *enr.Record
-	NodeID() enode.ID
-	DiscoveryAddresses() ([]multiaddr.Multiaddr, error)
-	RefreshPersistentSubnets()
-	FindPeersWithSubnet(ctx context.Context, topic string, subIndex uint64, threshold int) (bool, error)
-	AddPingMethod(reqFunc func(ctx context.Context, id peer.ID) error)
-}
-
-// Sender abstracts the sending functionality from libp2p.
-type Sender interface {
-	Send(context.Context, interface{}, string, peer.ID) (network.Stream, error)
-}
-
-// PeersProvider abstracts obtaining our current list of known peers status.
-type PeersProvider interface {
-	Peers() *peers.Status
-}
-
-// MetadataProvider returns the metadata related information for the local peer.
-type MetadataProvider interface {
-	Metadata() metadata.Metadata
-	MetadataSeq() uint64
-}
-
-type DataColumnsHandler interface {
-	CustodyGroupCountFromPeer(peer.ID) uint64
-}
+type (
+	P2P interface {
+		Broadcaster
+		SetStreamHandler
+		PubSubProvider
+		PubSubTopicUser
+		SenderEncoder
+		PeerManager
+		ConnectionHandler
+		PeersProvider
+		MetadataProvider
+		DataColumnsHandler
+	}
+
+	Acceser interface {
```

Comment:
```suggestion
	Accessor interface {
```

Want to write a godoc too?

Reply:
Fixed in 4bad0d864d4bc6473661cb3c8657901305dac320.

(Not accepting your suggestion as is because `Acceser` needs to be modified in some other places in the code as well.)